### PR TITLE
Add Docker build

### PR DESCRIPTION
**Pull request by vadi2**
#### Brief overview of PR changes/additions
Add a docker-compose build.
#### Motivation for adding to package repo
Easier setup - don't have to configure all of the dependencies manually. Maybe we can get rid of fakeSMTP and use mailgun for everything as well? The developer would need to setup their own mailgun account then.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
+
+## Via local development
 
 * Install [OpenResty](https://openresty.org/en/installation.html)
 * Install [Luarocks](https://github.com/luarocks/luarocks/wiki/Download)
 * `sudo apt install libssl-dev` (on Ubuntu)
 * `luarocks install lapis`
 * `luarocks install bcrypt`
 * `luarocks install i18n`
+* `luarocks install lua-mailgun`
```
Comment (Edru2) [inline README.md]:
```suggestion
* `luarocks install mailgun`
```

**Comment by Edru2** [comment]
I installed Docker and Docker-compose, but how to use it?
Maybe we should specify that in the readme.

**Comment by vadi2** [comment]
Yeah that's weird, I did add it there - not sure where it went again. It's there now, try again!

File: config.lua [hunk 2]
```diff
@@ -1,21 +1,35 @@
---config.lua
-local config = require("lapis.config")
-config({'development'}, {
+-- any environment variables used here must also be declared in nginx.conf
+
```
Comment (Edru2) [inline config.lua]:
```suggestion
local config = require("lapis.config")
```

**Comment by Edru2** [comment]
I have some  problems with getting the server to work with docker.

- I had to remove my postgresql installation to get it up, otherwise the installations collided somehow

- sometimes (cannot reproduce it every time) when I start docker (docker-compose up) the server won't start and I get ```mudlet-package-repo_package-repo_1 exited with code 1``` multiple times.
this is only solvable (some times) by deleting the data folder and rebuilding the docker (docker-compose build). I tried (as Vadi suggested) to change the data folder name for the database but that didn't really solve the problem.

- even if the server works in the docker environment I have problems with sending the email verification. When I register a new user I get an error ```connect failure: connection refused```
Edit: The email verification issue only occurs if the server is started in docker

**Comment by vadi2** [comment]
> I had to remove my postgresql installation to get it up, otherwise the installations collided somehow

If you remove the `ports` line for postgresql, does it fix the problem and does it still work?

**Comment by Edru2** [comment]
> If you remove the `ports` line for postgresql, does it fix the problem and does it still work?

That seems to solve the issue!

Edit:
Remove the mkdir data && part in Dockerfile seems to solve the other issue with ```mudlet-package-repo_package-repo_1 exited with code 1```

**Comment by Edru2** [comment]
>     * even if the server works in the docker environment I have problems with sending the email verification. When I register a new user I get an error `connect failure: connection refused`
>       Edit: The email verification issue only occurs if the server is started in docker

using mailgun also works in docker environment.

Notice that in order to get mailgun to work it is necessary to add username and password to mail.lua in mail_handler.new part

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+version: '3.1'
+
+services:
+
+  # PostgreSQL database server
+  psql:
+    image: postgres
+    ports:
+      - 5432:5432
```
Comment (Edru2) [inline docker-compose.yml]:
```suggestion
```
Makes it able to use already installed postgres together with docker postgres at the same machine

File: Dockerfile [hunk 4]
```diff
@@ -26,4 +28,4 @@ COPY . .
 
 EXPOSE 8080
 
-CMD [ "lapis", "server" ]
+CMD mkdir data && lapis migrate $LAPIS_ENVIRONMENT && lapis server $LAPIS_ENVIRONMENT
```
Comment (Edru2) [inline Dockerfile]:
```suggestion
CMD lapis migrate $LAPIS_ENVIRONMENT && lapis server $LAPIS_ENVIRONMENT
```

**Review by Edru2 — CHANGES_REQUESTED** [review]

**Review by demonnic — APPROVED** [review]

**Pull request merged**